lib/cretonne/src/dominator_tree.rs

```diff
@@ -323,6 +329,72 @@ impl DominatorTree {
     }
 }
 
+impl DominatorTree {
+    /// When splitting an `Ebb` using `Layout::split_ebb`, you can use this method to update
+    /// the dominator tree locally rather than recomputing it.
+    ///
+    /// `old_ebb` is the `Ebb` before splitting, and `new_ebb` is the `Ebb` which now contains
+    /// the second half of `old_ebb`.
+    pub fn recompute_split_ebb(&mut self, old_ebb: Ebb, new_ebb: Ebb, split_jump_inst: Inst) {
+        let old_ebb_postorder_index =
+            self.postorder
+            .as_slice()
+            // We use the RPO comparison on the postorder list so we invert the operands of the
+            // comparison
+            .binary_search_by(|probe| self.rpo_cmp_ebb(old_ebb,*probe,))
+            .expect("the old ebb is not declared to the dominator tree");
+        let new_ebb_rpo = self.insert_after_rpo(old_ebb, old_ebb_postorder_index);
+        *self.nodes.ensure(new_ebb) = DomNode {
+            rpo_number: new_ebb_rpo,
+            idom: Some(split_jump_inst).into(),
+        };
+        // TODO: insert in constant time?
+        self.postorder.insert(0, new_ebb);
+    }
+
+    // We want to insert a new ebb just after ref_ebb in the rpo order. This function checks
+    // if there is a gap in rpo numbers; if yes it returns the number in the gap and if
+    // not it renumbers.
+    fn insert_after_rpo(&mut self, ebb: Ebb, ebb_postorder_index: usize) -> u32 {
+        let ebb_rpo_number = self.nodes[ebb].rpo_number;
+        if ebb_postorder_index == 0 {
+            ebb_rpo_number + STRIDE
+        } else {
+            let prev_postorder_ebb = self.postorder[ebb_postorder_index - 1];
+            let prev_postorder_ebb_rpo_number = self.nodes[prev_postorder_ebb].rpo_number;
+            if prev_postorder_ebb_rpo_number > ebb_rpo_number + 1 {
+                // There is a gap, we can use it
+                ebb_rpo_number + 1
+            } else {
+                // We have to renumber
+                let return_value = ebb_rpo_number + 1;
```

Comment:
Both arms of this 'if' appear to be returning `ebb_rpo_number + 1` which could be simplified.